pkg/crc/segment/segment.go

```diff
@@ -52,7 +52,7 @@ func (c *Client) Close() error {
 	return c.segmentClient.Close()
 }
 
-func (c *Client) Upload(err error) error {
+func (c *Client) Upload(action string, err error) error {
```

Comment:
@guillaumerose This should be part of a different commit, here we are changing this method signature.So in this commit the `segmentMiddleware` just add the error not the action and then in the next commit you can add the action.

Reply:
done